api/v1/installation_types.go

```diff
@@ -435,6 +436,7 @@ type CalicoNetworkSpec struct {
 	// IPPools contains a list of IP pools to create if none exist. At most one IP pool of each
 	// address family may be specified. If omitted, a single pool will be configured if needed.
 	// +optional
+	// +kubebuilder:validation:MaxItems=25
```

Comment:
I think this is probably an OK limitation to put in to start? TBD. Most clusters shouldn't need 25+ pools. Better to start somewhat strict and relax later if the need arises?

Reply:
My original motivation was that I wanted to add more powerful kubebuilder validation here - e.g., uniqueness validaiton - but that it was rejected when the slice was boundless as too complicated of a validation rule.

I ended up not doing that for other reasons, but I kept this around since it's a hint to users that they might be doing something wrong if they need lots of IP pools (which is usually the case, but not necessarily always).

Maybe I should just remove it.

Reply:
I'd personally err on the side of caution, we don't want to cause problems for some customer who has some use case, however odd the use case is, for having a very large number of IPPools. I'd say it's probably only worth setting the cap if there is some bound to which we or kubernetes cannot reasonably handle more than that maximum amount.

Reply:
@Brian-McM maybe to reframe this - today, we have a hard limit of 2 IP pools. This PR increases that limit to 25.

Does the fact that this is a 10x increase in the number of pools you can specify change how you feel about this at all?

I can remove the limitation, but once it's removed there's no adding it back without breaking users.

Reply:
> Does the fact that this is a 10x increase in the number of pools you can specify change how you feel about this at all?

Oh absolutely, I some how wasn't realising / remembering that you could only specify two before, I thought this somehow affected that cap on all IP pool in the cluster (which is ridiculous).

Seems like a good idea to cap it 😀!